File: Packs/CommonScripts/Scripts/GetServerURL/GetServerURL.py
from urllib.parse import ParseResult, urlparse

import demistomock as demisto
from CommonServerPython import *


def main():
    urls = demisto.demistoUrls()
    server_url = urls.get("server", "")

    server_url_parts: ParseResult = urlparse(server_url)
    host_name = server_url_parts.hostname
    port = server_url_parts.port
    scheme = server_url_parts.scheme

    server_address = {"Scheme": scheme, "Host": host_name, "Port": port, "URL": server_url}
    return_results(
        CommandResults(
            readable_output=server_url,
            outputs=server_address,
            raw_response=server_url,
            outputs_prefix="ServerURL",
            outputs_key_field="URL",
        )
    )


if __name__ in ("__builtin__", "builtins"):
    main()

File: Packs/CommonScripts/Scripts/GetServerURL/GetServerURL.js
  function parseUrl(url) {
      match = url.match(urlRegex);

      return match ? {
          shcema: url.match(/^(https?):/)[1],
          hostname: match[2],
          port: match[3] || '',
      } : null;
  }

  urls = demistoUrls();
  server_url = urls['server'].toString();


  result = parseUrl(server_url);

  ec = {
      'ServerURL': {
          Scheme: result.shcema,
          Host: result.hostname,
          Port: Number(result.port),
          URL: server_url
      }
  }

  return {
      Type: entryTypes.note,
      Contents: server_url,
      ContentsFormat: formats.note,
      HumanReadable: server_url,
      ReadableContentsFormat: formats.markdown,
      EntryContext: ec
  };